pkg/schedule/schedulers/balance_range.go

```diff
@@ -372,43 +372,114 @@ type balanceRangeScheduler struct {
 	expectScoreMap map[uint64]float64
 	job            *balanceRangeSchedulerJob
 	solver         *solver
+	lastReportTime time.Time
+}
+
+func (s *balanceRangeScheduler) report() {
+	now := time.Now()
+	if now.Sub(s.lastReportTime) < 10*time.Second {
+		return
+	}
+	for storeID, score := range s.scoreMap {
+		storeStr := strconv.FormatUint(storeID, 10)
+		balanceRangeGauge.WithLabelValues(storeStr, "score").Set(score)
+		if expect, ok := s.expectScoreMap[storeID]; ok {
+			balanceRangeGauge.WithLabelValues(storeStr, "expect").Set(expect)
+		} else {
+			balanceRangeGauge.WithLabelValues(storeStr, "expect").Set(0)
+		}
+	}
+	balanceRangeJobGauge.WithLabelValues().Set(float64(s.job.JobID))
+	s.lastReportTime = now
 }
 
 // ServeHTTP implements the http.Handler interface.
 func (s *balanceRangeScheduler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	s.handler.ServeHTTP(w, r)
 }
 
+func (s *balanceRangeScheduler) isBalanced() bool {
+	diff := 0
+	for storeID, score := range s.scoreMap {
+		if expect, ok := s.expectScoreMap[storeID]; ok {
+			if expect > score {
+				diff += int(expect - score)
+			} else {
+				diff += int(score - expect)
+			}
+		}
+	}
+	return diff <= 1
+}
+
 // IsScheduleAllowed checks if the scheduler is allowed to schedule new operators.
 func (s *balanceRangeScheduler) IsScheduleAllowed(cluster sche.SchedulerCluster) bool {
 	allowed := s.OpController.OperatorCount(operator.OpRange) < cluster.GetSchedulerConfig().GetRegionScheduleLimit()
 	if !allowed {
 		operator.IncOperatorLimitCounter(s.GetType(), operator.OpRange)
+		return false
 	}
 	if err := s.conf.gc(); err != nil {
```

Comment:
Shall we move the following code to the Schedule function?

Reply:
good suggestion, I will move this codes to new function